src/codeSigning/packAndSign.ts

```diff
@@ -166,35 +171,36 @@ export const api = {
    * @param publicKeyUrl - url for the public key
    */
   verify(tarGzStream: Readable, sigFilenameStream: Readable, publicKeyUrl: string): Promise<boolean> {
-    return new Promise<boolean>((resolve, reject) => {
+    return new Promise<boolean>((resolve) => {
       const verifyInfo = new CodeVerifierInfo();
       verifyInfo.dataToVerify = tarGzStream;
       verifyInfo.signatureStream = sigFilenameStream;
 
-      const req = https.get(publicKeyUrl, { agent: false });
-
-      req.on('response', (response) => {
-        if (response && response.statusCode === 200) {
-          verifyInfo.publicKeyStream = response;
-          return resolve(verify(verifyInfo));
-        } else {
-          const statusCode: number = response.statusCode;
-          return reject(
-            new NamedError(
-              'RetrievePublicKeyFailed',
-              `Couldn't retrieve public key at url: ${publicKeyUrl} error code: ${statusCode}`
-            )
-          );
-        }
-      });
-
-      req.on('error', (err: Error) => {
-        if (err && err['code'] === 'DEPTH_ZERO_SELF_SIGNED_CERT') {
-          reject(new SignSignedCertError());
-        } else {
-          reject(err);
-        }
-      });
+      return resolve(
+        (async (): Promise<boolean> => {
+          try {
+            const agent = api.getAgentForUri(publicKeyUrl);
+            const response = await got.get(publicKeyUrl, { agent });
+            if (response && response.statusCode === 200) {
+              verifyInfo.publicKeyStream = Readable.from([response.body]);
+              return await verify(verifyInfo);
+            } else {
+              const statusCode: number = response.statusCode;
+              throw new NamedError(
+                'RetrievePublicKeyFailed',
+                `Couldn't retrieve public key at url: ${publicKeyUrl} error code: ${statusCode}`
+              );
+            }
+          } catch (err) {
+            const error = err as RequestError;
+            if (error && error.code === 'DEPTH_ZERO_SELF_SIGNED_CERT') {
+              throw new SignSignedCertError();
```

Comment:
I will try to use reject. Was having issues with tests failing and using throw fixed the tests.